src/button-dropdown/internal.tsx

```diff
@@ -77,9 +77,6 @@ const InternalButtonDropdown = React.forwardRef(
     useForwardFocus(ref, dropdownRef);
 
     const clickHandler = () => {
-      if (!usingMouse.current) {
-        return;
-      }
```

Comment:
Remove it because it is for preventing firing onClick event when Space/Enter is pressed on IE11.  CR-35629483